### PR TITLE
Add support for HDD Storage Type

### PR DESCRIPTION
**Pull request by gkao123**
**Is this a bug fix or adding new feature?**
Adding support for HDD, a new FSx feature

https://aws.amazon.com/about-aws/whats-new/2020/08/amazon-fsx-lustre-announces-high-performance-hdd-based-shared-storage-compute-workloads/

**What testing is done?** 
End to end testing

### CODE REVIEW

**Comment by gkao123** [comment]
@leakingtapan Hi Cheng, the tests are failing due to the following error message:
```failed to created filesystem There is already a disk with same ID and different size```.

Could this be related to previous failed test runs?

**Comment by wongma7** [comment]
I think the bug has to do with how we resuse the same fsid forall tests to avoid hitting the 5 minute creation timeout, but we should not need to anymore ever since https://github.com/kubernetes/kubernetes/pull/84140 merged

**Comment by wongma7** [comment]
Never mind, I am not sure the issue, because actually it is the pre-provisioned tests (these tests are SUPPOSED to resuse the same volume) that are failing. will keep digging.

**Comment by wongma7** [comment]
OK, I checked cloudtrail and the error there does not match what we got back?

```
    "errorCode": "IncompatibleParameterError",
    "errorMessage": "The cache type can not be specified for a lustre file system with DeploymentType=SCRATCH_1",
```

Never mind about the volume reuse stuff, since BOTH dynamic provisioning and pre-provisioned tests make a CreateVolume call, and the reuse issue was fixed a long time ago already, that can't be it even though the error message "
      failed to created filesystem There is already a disk with same ID and different size " sounds a lot like it.

File: tests/e2e/conformance_test.go [hunk 1]
```diff
@@ -93,12 +93,16 @@ func (e *fsxDriver) CreateVolume(config *testsuites.PerTestConfig, volType testp
...
 	subnetId := *instance.SubnetId
+	defaultStorageType := "HDD"
+	defaultDriveCacheType := "NONE"
 
 	ctx := context.Background()
 	options := &fsx.FileSystemOptions{
 		CapacityGiB:      3600,
 		SubnetId:         subnetId,
 		SecurityGroupIds: securityGroupIds,
+		StorageType:      defaultStorageType,
```
Comment (wongma7) [inline tests/e2e/conformance_test.go:104]: per my latest comment including the cloudtrail message, also need to set deploymentType to PERSISTENT_1?

Comment (gkao123) [reply tests/e2e/conformance_test.go]: I included this line of code due to a previous test failure. I realize now that the previous test failure was related to a bug in another code path, so I've gone ahead and removed this parameter in the latest revision.

**Pull request merged**